pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -98,35 +99,32 @@ func (p *PoolManager) AllocateVirtualMachineMac(virtualMachine *kubevirt.Virtual
 	return nil
 }
 
-func (p *PoolManager) ReleaseVirtualMachineMac(vm *kubevirt.VirtualMachine) error {
+func (p *PoolManager) ReleaseVirtualMachineMac(vm *kubevirt.VirtualMachine, reconcileLog logr.Logger) error {
+	logger := reconcileLog.WithName("ReleaseVirtualMachineMac")
+
 	p.poolMutex.Lock()
 	defer p.poolMutex.Unlock()
-	log.V(1).Info("ReleaseVirtualMachineMac: data",
+	logger.V(1).Info("data",
 		"macmap", p.macPoolMap,
 		"podmap", p.podToMacPoolMap,
 		"currentMac", p.currentMac.String())
 
 	if len(vm.Spec.Template.Spec.Domain.Devices.Interfaces) == 0 {
-		log.V(1).Info("no interfaces found for virtual machine, skipping mac release",
```

Comment:
Here the passed logger starts to be a little weird. We assume that it contains VM info, so we are dropping it from here. But there is no guarantee somebody removes it from the parent logger. I'm puzzled. I don't think we can do anything about it. I don't like the fact that it semantically glues the whole chain together :grimacing: Any suggestion?

Reply:
well for better and for worse, the reason we did this is because there is information (the unique id, the vm info) that we hold in the first reconcile function, that we want to remember while reading the logs but don't necessarily want to carry with us to each future function. 
But I think that since the both (the vm info and the id) are consistent and always relevant through out the Reconcile function - then it's ok. what do you think?

Reply:
I hate that we depend on implicitly given context, but I don't have a better option to propose. So I guess it's ok